src/core/Fingerprint.cpp

```diff
@@ -0,0 +1,240 @@
+#include "Fingerprint.hpp"
+#include "../helpers/Log.hpp"
+#include "src/config/ConfigManager.hpp"
+
+#include <filesystem>
+#include <unistd.h>
+#include <pwd.h>
+
+#include <cstring>
+#include <thread>
+
+static const auto FPRINT        = "net.reactivated.Fprint";
+static const auto DEVICE        = "net.reactivated.Fprint.Device";
+static const auto MANAGER       = "net.reactivated.Fprint.Manager";
+static const auto LOGIN_MANAGER = "org.freedesktop.login1.Manager";
+static const auto RETRY_MESSAGE = "Could not match fingerprint. Try again.";
+
+CFingerprint::CFingerprint() {
+    static auto* const PFINGERPRINTREADY   = (Hyprlang::STRING*)(g_pConfigManager->getValuePtr("general:fingerprint_ready_message"));
+    m_sFingerprintReady                    = *PFINGERPRINTREADY;
+    static auto* const PFINGERPRINTPRESENT = (Hyprlang::STRING*)(g_pConfigManager->getValuePtr("general:fingerprint_present_message"));
+    m_sFingerprintPresent                  = *PFINGERPRINTPRESENT;
+}
+
+static void fingerprintUnlockCallback(std::shared_ptr<CTimer> self, void* data) {
+    g_pHyprlock->unlock();
```

Comment:
I don't really like that, because i think the check for `m_bAuthenticated` should be right here. Just make it a `onFingerprintCheckTimer` like with the password authentication and call it whether or not we are authenticated.

Reply:
Done